### DISCUSSION TITLE
Shareable Routes?

### DISCUSSION THREAD

**Discussion by idleberg** [sveltejs/kit#5880] [Q&A]
Since SvelteKit is a file-based router, I was wondering if there is some way to make routes shareable. I have a very specific use case for that myself, but let's use a more approachable example instead.

Let's say there is this awesome library for error pages, it's really descriptive, internationalized, and whatnot. It's attractive to use and people want to use it for their error routes. From my understanding, I would still have to create multiple routes and import modules in each of them.

Is this correct or is there an alternative I don't know about yet? I was hoping that [advanced routing](https://kit.svelte.dev/docs/routing#advanced-routing) might offer something to solve this, but I don't see how.

**Comment by vekunz** [comment]
We would have a use case for shareable routes also. We are implementing several services with shared code. The shared code is shared through local npm packages. Currently, we have eight standalone endpoints and one svelte page (actually an error page ;) ) that are identical through every service.
For some of the standalone endpoints we have moved the code to the library and just reexport them in the SvelteKit app in the specific route file. For the svelte route, we did not find a good solution to move the code to the library. It would be great for both the standalone endpoint and the svelte pages to move them completely to a library, without needing a file for every route in the SvelteKit app.

**Comment by lukaszpolowczyk** [comment]
Related:

 Adder ability, to add Hooks handle in sequence 
https://github.com/sveltejs/kit/issues/3723

External Named Layout - from package 
https://github.com/sveltejs/kit/issues/4518

**Comment by lukaszpolowczyk** [comment]
Theoretically loosely related (you have routes from the package, but not selected, but all at once):
https://twitter.com/joshnuss/status/1553580203540094977

**Comment by danielbayley80** [comment]
Currently looking for a solution to this. I have things like user modules, payment modules, etc.
With express I can reference them and plug them in to the express pipeline. 
I was trying to work out how to do this in svelte.

Example :
```
      const app = express()   ;

      const userOptions = new UserModuleOptions();
      UserModule.initialise(app, mongoClient, userOptions);

      const cmsOptions = new CmsModuleOptions();
      CmsModule.initialiseStatic(express, app, mongoClient,  cmsOptions);

```

**Comment by gregg-cbs** [comment]
My case is that I need a page to be available at more then one route.

I have a page with server code at `/dashboard/events/12/orders` but I also want the page to be available at `/orders`. Right now I am copying and pasting the files because of file based routing. Not sure if there is a solution to this.

**Reply by PudottaPommin** [reply]
Try checking [reroute](https://kit.svelte.dev/docs/hooks#universal-hooks-reroute) in recent version of kit

**Reply by gregg-cbs** [reply]
Thanks for that it could solve my issue. When reading that documentation page i see the mention of hooks... is hooks not a silly name for middleware? and a silly name in that its easily confused with react hooks but not at all the same thing and it works like middleware but its called something else? I actually thought it was lifecycle events.
